pkg/volume/glusterfs/glusterfs.go

```diff
@@ -302,3 +337,170 @@ func getVolumeSource(
 
 	return nil, false, fmt.Errorf("Spec does not reference a Gluster volume type")
 }
+
+func (plugin *glusterfsPlugin) NewProvisioner(options volume.VolumeOptions) (volume.Provisioner, error) {
+	if len(options.AccessModes) == 0 {
+		options.AccessModes = plugin.GetAccessModes()
+	}
+	return plugin.newProvisionerInternal(options)
+}
+
+func (plugin *glusterfsPlugin) newProvisionerInternal(options volume.VolumeOptions) (volume.Provisioner, error) {
+	return &glusterfsVolumeProvisioner{
+		glusterfsMounter: &glusterfsMounter{
+			glusterfs: &glusterfs{
+				plugin: plugin,
+			},
+		},
+		options: options,
+	}, nil
+}
+
+type glusterfsClusterConf struct {
+	Glusterep          string `json:"endpoint"`
+	GlusterRestvolpath string `json:"path"`
+	GlusterRestUrl     string `json:"resturl"`
+	GlusterRestAuth    bool   `json:"restauthenabled"`
+	GlusterRestUser    string `json:"restuser"`
+	GlusterRestUserKey string `json:"restuserkey"`
+}
+
+type glusterfsVolumeProvisioner struct {
+	*glusterfsMounter
+	*glusterfsClusterConf
+	options volume.VolumeOptions
+}
+
+func (plugin *glusterfsPlugin) NewDeleter(spec *volume.Spec) (volume.Deleter, error) {
```

Comment:
umm, if all the configuration  parameters are in StorageClass, then `Deleter` does not have access to it. I'll add `Parameters` argument here when DPv2 is merged.